lib/widgets/molecules/recommended_mentors_filters.dart

```diff
@@ -0,0 +1,251 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:textfield_tags/textfield_tags.dart';
+import 'package:mm_flutter_app/constants/app_constants.dart';
+import 'package:mm_flutter_app/widgets/atoms/autocomplete_widget.dart';
+import 'package:mm_flutter_app/providers/explore_card_filters_provider.dart';
+
+class RecommendedMentorsFilters extends StatefulWidget {
+  final ExploreCardFiltersProvider filtersProvider;
+
+  const RecommendedMentorsFilters({super.key, required this.filtersProvider});
+
+  @override
+  State<StatefulWidget> createState() => _RecommendedMentorsFilters();
+}
+
+class _RecommendedMentorsFilters extends State<RecommendedMentorsFilters> {
+  TextfieldTagsController _countriesController = TextfieldTagsController();
+  TextfieldTagsController _languagesController = TextfieldTagsController();
+  Set<String> _selectedSkills = {};
+
+  @override
+  void initState() {
+    super.initState();
+    _countriesController = TextfieldTagsController();
+    _languagesController = TextfieldTagsController();
+    _selectedSkills = widget.filtersProvider.selectedSkills;
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _countriesController.dispose();
+    _languagesController.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final AppLocalizations l10n = AppLocalizations.of(context)!;
+
+    return Scaffold(
+      appBar: AppBar(title: Text(l10n.exploreSearchFilterTitle)),
```

Comment:
I've added this widget as the topmost route inside `AppRouter`. It still renders a `Scaffold` on its own but skips the one from `AppWrapper`. This seems beneficial to me as this `Scaffold` is rather simple (only `appBar` is set) and is independent from the application state.

Reply:
Hmm the problem is that now the RecommendedMentorsFilters is outside the ShellRoute, meaning that it is not part of the same navigation stack. The entire navigation stack is lost the moment you get out of the ShellRoute. Try pressing the 'back' button on the filters page and it will exit the app instead of navigating to the previous screen.

Reply:
I didn't notice that. This is a good reason to have this inside `ShellRoute`. Thanks.

I've changed it appropriately but it seems the "back" button still doesn't work. :| Could you please take a look at it?

Reply:
Turns out I was wrong. The ShellRoute does have its own navigation stack, but it wasn't the cause behind the app closing when back is pressed. Back navigation seems to break we use context.go instead of context.push (or pop). I replaced the calls in this code to use push to remove the issue, although I would like to investigate more why using go causes such big problems.

I will merge now, thanks for bearing with me until the end 😅